### ISSUE TITLE
AttributeError while running a workflow tutorial 

### ISSUE THREAD

**Issue by athanell** [nipy/nipype#2242]
### Summary
While running the Workflow tutorial (https://miykael.github.io/nipype_tutorial/notebooks/introduction_quickstart.html)
I get into an AttributeError while executing the eg = wf.run()

### Script/Workflow details

```
import os
from nipype import Workflow, Node, Function

def sum(a, b):
    return a + b

wf = Workflow('hello')

adder = Node(Function(input_names=['a', 'b'],
                      output_names=['sum'],
                      function=sum), 
             name='a_plus_b')

adder.inputs.a = 1
adder.inputs.b = 3

wf.add_nodes([adder])

wf.base_dir = os.getcwd()

eg = wf.run()
```

### Error output

```
 eg = wf.run()
171020-18:41:18,716 workflow INFO:
	 Workflow hello settings: ['check', 'execution', 'logging']
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-3330efeb744d> in <module>()
----> 1 eg = wf.run()

~/anaconda3/envs/SAH/lib/python3.5/site-packages/nipype/pipeline/engine/workflows.py in run(self, plugin, plugin_args, updatehash)
    577             del self.config['crashdump_dir']
    578         logger.info('Workflow %s settings: %s', self.name, to_str(sorted(self.config)))
--> 579         self._set_needed_outputs(flatgraph)
    580         execgraph = generate_expanded_graph(deepcopy(flatgraph))
    581         for index, node in enumerate(execgraph.nodes()):

~/anaconda3/envs/SAH/lib/python3.5/site-packages/nipype/pipeline/engine/workflows.py in _set_needed_outputs(self, graph)
    670         for node in graph.nodes():
    671             node.needed_outputs = []
--> 672             for edge in graph.out_edges_iter(node):
    673                 data = graph.get_edge_data(*edge)
    674                 sourceinfo = [v1[0] if isinstance(v1, tuple) else v1

AttributeError: 'DiGraph' object has no attribute 'out_edges_iter'
```

### Platform details:

```
{'scipy_version': '0.19.1', 'commit_hash': '%h', 'pkg_path': '/home/athanell/anaconda3/envs/SAH/lib/python3.5/site-packages/nipype', 'nibabel_version': '2.1.0', 'sys_platform': 'linux', 'numpy_version': '1.13.3', 'commit_source': 'archive substitution', 'sys_executable': '/home/athanell/anaconda3/envs/SAH/bin/python', 'networkx_version': '2.0', 'nipype_version': '0.13.1', 'sys_version': '3.5.4 |Anaconda, Inc.| (default, Oct  5 2017, 08:00:22) \n[GCC 7.2.0]', 'traits_version': '4.6.0'}
0.13.1
```

Any ideas on what is causing this error?
Thank you

**Comment by effigies** [comment]
You may need to update your `networkx` package. What version do you have? We may need to update the minimum version.

**Comment by oesteban** [comment]
@djarecka probably knows the exact answer :)

**Comment by athanell** [comment]
Hello @effigies , 
The version I use is 2.0

**Comment by effigies** [comment]
Ah, okay. I got the issue backwards. With 2.0, the old `_iter` interfaces were dropped. I'd recommend installing from master.

```Shell
pip install --user git+https://github.com/nipy/nipype.git@master#egg=nipype-0.14.0-dev
```

**Comment by athanell** [comment]
Hi again @effigies , 
I installed with pip from master, and the command
`eg = wf.run()`
does not fail but the following does
`eg.nodes()[0].result.outputs`

and the error in the last one is 

```
In [12]: eg.nodes()[0].result.outputs
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-12-88eca3d662c4> in <module>()
----> 1 eg.nodes()[0].result.outputs

~/anaconda3/envs/SAH/lib/python3.5/site-packages/networkx/classes/reportviews.py in __getitem__(self, n)
    176 
    177     def __getitem__(self, n):
--> 178         return self._nodes[n]
    179 
    180     # Set methods

KeyError: 0
```

**Comment by effigies** [comment]
Hmm. Looks like perhaps `eg.nodes()` has a `dict`-like interface, instead of a `list`-like. Maybe try `list(eg.nodes())` and see what that looks like.

**Comment by athanell** [comment]
Yes @effigies ,
your suggestion resolved the error. 
Thank you for the assistance

**Comment by djarecka** [comment]
this will come back... will look into it.

**Comment by djarecka** [comment]
I just reviewed my PR, and I also had to change `eg.nodes` to list in a few places, e.g. [here](https://github.com/nipy/nipype/blob/master/nipype/pipeline/plugins/tests/test_linear.py#L44),
since `eg` is jus a `networkx.classes.digraph.DiGraph` object and the `node` method has changed.

However, this is syntax that is used by many users and they would have to introduce this small change everywhere. Unless, we want to create our own class and overwrite the `node` method
